core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -879,70 +874,60 @@ object IO {
+   * return the results in a new `List[B]`. For parallelism use `parTraverse`.
    */
-  final def traverse[E, A, B, M[X] <: TraversableOnce[X]](
-    in: M[A]
-  )(fn: A => IO[E, B])(implicit cbf: CanBuildFrom[M[A], B, M[B]]): IO[E, M[B]] =
-    in.foldLeft[IO[E, mutable.Builder[B, M[B]]]](IO.sync(cbf(in)))((io, b) => io.zipWith(fn(b))(_ += _))
-      .map(_.result())
+  final def traverse[E, A, B](in: List[A])(fn: A => IO[E, B]): IO[E, List[B]] =
+    in.foldRight[IO[E, List[B]]](IO.sync(Nil)) { (a, io) =>
+      io.zipWith(fn(a))((bs, b) => b :: bs)
+    }
 
   /**
-   * Evaluate the elements of a traversable data structure in parallel
+   * Evaluate the elements of a `List[A]` in parallel
    * and collect the results. This is the parallel version of `traverse`.
    */
-  def parTraverse[E, A, B, M[X] <: TraversableOnce[X]](
-    as: M[A]
-  )(fn: A => IO[E, B])(implicit cbf: CanBuildFrom[M[A], B, M[B]]): IO[E, M[B]] =
-    as.foldLeft[IO[E, mutable.Builder[B, M[B]]]](IO.sync(cbf(as))) { (bsIO, a) =>
-        bsIO.par(fn(a)).map {
-          case (bs, b) => bs += b
-        }
-      }
-      .map(_.result)
+  def parTraverse[E, A, B](as: List[A])(fn: A => IO[E, B]): IO[E, List[B]] =
+    as.foldRight[IO[E, List[B]]](IO.sync(Nil)) { (a, io) =>
+      io.par(fn(a)).map { case (bs, b) => b :: bs }
+    }
 
   /**
    * Evaluate each effect in the structure from left to right, and collect
    * the results. For parallelism use `parAll`.
    */
-  final def sequence[E, A, M[X] <: TraversableOnce[X]](
-    in: M[IO[E, A]]
-  )(implicit cbf: CanBuildFrom[M[IO[E, A]], A, M[A]]): IO[E, M[A]] =
+  final def sequence[E, A](in: List[IO[E, A]]): IO[E, List[A]] =
     traverse(in)(identity)
 
   /**
    * Evaluate each effect in the structure in parallel, and collect
    * the results. This is the parallel version of `sequence`.
    */
-  final def parAll[E, A, M[X] <: TraversableOnce[X]](
-    as: M[IO[E, A]]
-  )(implicit cbf: CanBuildFrom[M[IO[E, A]], A, M[A]]): IO[E, M[A]] =
+  final def parAll[E, A](as: List[IO[E, A]]): IO[E, List[A]] =
     parTraverse(as)(identity)
 
   /**
-   * Races a traversable collection of `IO[E, A]` against each other. If all of
+   * Races a list of `IO[E, A]` against each other. If all of
    * them fail, the last error is returned.
    *
    * _Note_: if the collection is empty, there is no action that can either
    * succeed or fail. Therefore, the only possible output is an IO action
    * that never terminates.
    */
-  final def raceAll[E, A](t: TraversableOnce[IO[E, A]]): IO[E, A] =
+  final def raceAll[E, A](t: List[IO[E, A]]): IO[E, A] =
     t.foldLeft[IO[E, A]](IO.terminate(NothingRaced))(_ race _)
 
   /**
    * Reduces a list of IO to a single IO, works in parallel.
    */
-  final def reduceAll[E, A](a: IO[E, A], as: TraversableOnce[IO[E, A]])(f: (A, A) => A): IO[E, A] =
+  final def reduceAll[E, A](a: IO[E, A], as: List[IO[E, A]])(f: (A, A) => A): IO[E, A] =
```

Comment:
Returning a `List` seems sensible, but there is no cost to accept a `Iterable`, which exists < 2.12 and >= 2.13.